libraries/AP_InertialSensor/AP_InertialSensor_Backend.h

```diff
@@ -252,10 +266,15 @@ class AP_InertialSensor_Backend
     // support for updating filter at runtime
     uint16_t _last_accel_filter_hz[INS_MAX_INSTANCES];
     uint16_t _last_gyro_filter_hz[INS_MAX_INSTANCES];
-    uint16_t _last_notch_center_freq_hz[INS_MAX_INSTANCES];
-    uint16_t _last_notch_bandwidth_hz[INS_MAX_INSTANCES];
+    float _last_notch_center_freq_hz[INS_MAX_INSTANCES];
+    float _last_notch_bandwidth_hz[INS_MAX_INSTANCES];
     float _last_notch_attenuation_dB[INS_MAX_INSTANCES];
 
+    // support for updating dynamic filter at runtime
+    float _last_harmonic_notch_center_freq_hz[INS_MAX_INSTANCES];
+    float _last_harmonic_notch_bandwidth_hz[INS_MAX_INSTANCES];
+    float _last_harmonic_notch_attenuation_dB[INS_MAX_INSTANCES];
```

Comment:
This is wrong (as is for all the above). This is in the Backend class so it already is per-instance, no array needed. You should fix the code adding this and if you can make a **separate commit** fixing the other ones I'd be grateful!

Reply:
So I agree that this is the backend, but every single "last_" member variable follows this pattern (not written by me). So probably all of them should be changed or none of them - I just copied what was there already. What would you like me to do? Oh I think I see - you want all of them fixed but the ones not added by me in a separate commit?

Reply:
> Oh I think I see - you want all of them fixed but the ones not added by me in a separate commit?

Yes, that's it. Separate commit is for easily noting there was a previous bug, not introduced by you. As said, you doing commit is optional as the code is already there and you aren't forced to fix it - you are forced to fix your new code though 🙂